server/routes/ErrorHandlingMiddleware.js

```diff
@@ -0,0 +1,23 @@
+function ErrorHandler(err, req, res, next) {
+  console.error("Error", err);
+  res.status(500).json({
```

Comment:
I think this function will always set 500 status error which indicates server error but there can be other errors too like 404 or 401

Reply:
Great catch! I created a HandledError as a subclass of Error so we can pass error codes.